queries/haskell/highlights.scm

```diff
@@ -11,6 +23,31 @@
 
 (comment) @comment
 
+; FIXME: The below documentation comment queries are inefficient
+; and need to be anchored, using something like
+; ((comment) @_first . (comment)+ @comment.documentation)
+; once https://github.com/neovim/neovim/pull/24738 has been merged.
+;
+; ((comment) @comment.documentation 
+;   (#lua-match? @comment.documentation "^-- |"))
```

Comment:
This query is fine, you can add it if you want.

Reply:
Ah, so the usage of this is supposed to be with the quantifiers.
Just ignore my comment and the one after